core-curriculum/knowledge/scala.json

```diff
@@ -170,104 +342,431 @@
-      "dependencies": ["collections-ii"]
+      "dependencies": ["collections-ii"],
+      "resources": [
+        {
+          "name": "ScalaDoc: The Architecture of Scala Collections",
+          "type": "article",
+          "url": "http://docs.scala-lang.org/overviews/core/architecture-of-scala-collections.html",
+          "tags": ["official"]
+        }
+      ]
     },
     {
       "id": "streams",
       "name": "Streams",
       "description": "Streams allow us to work with potentially infinite data structures.",
-      "dependecies": ["collections-i", "functions-as-values", "laziness"]
+      "dependecies": ["collections-i", "functions-as-values", "laziness"],
+      "resources": [
+        {
+          "name": "An extended example: lazy lists",
+          "type": "book",
+          "notes": "Chapter 5: 'Strictness and lazyness', section 5.2",
+          "url": "https://www.manning.com/books/functional-programming-in-scala"
+        },
+        {
+          "name": "Understanding Scala streams through Finbonacci",
+          "type": "article",
+          "url": "http://derekwyatt.org/2011/07/29/understanding-scala-streams-through-fibonacci/"
+        }
+      ],
+
+      "abilities": [
+        "Code your own simple Stream, using a thunk to generate them indefinitely."
+      ],
+
+      "assessment-questions": [
+        { "question": "Why do we use thunk or by-name arguments in lazy structures? Why can we just use strict evaluation?" },
+        { "question": "Normally, what happens with the data that already has been processed over a stream? Why? Which are the pros and cons?" }
+      ]
     },
     {
       "id": "variance",
       "name": "Variance",
       "description": "With type parameters and type constructors comes variance, which defines for a give type A that extends B, how is the relationship of C[A] and C[B], given a type constructor C.",
       "tags": ["required-for-scala-senior"],
-      "dependencies": ["type-parameters", "type-constructors", "collections-i"]
+      "dependencies": ["type-parameters", "type-constructors", "collections-i"],
+      "resources": [
+        {
+          "name": "Wikipedia: Covariance and contravariance",
+          "type": "article",
+          "url": "https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)",
+          "tags": ["wikipedia"]
+        },
+        {
+          "name": "Variance",
+          "type": "video",
+          "url": "https://www.coursera.org/learn/progfun1/lecture/dnreZ/lecture-4-4-variance-optional",
+          "tags": ["coursera", "mooc"]
+        },
+        {
+          "name": "The Scala Type System: Parameterized Types and Variances, Part 1",
+          "type": "article",
+          "url": "https://blog.codecentric.de/en/2015/03/scala-type-system-parameterized-types-variances-part-1/"
+        }
+      ],
+
+      "abilities": [
+        "Recognize what's to be in covariant or contravariant position",
```

Comment:
Suggestion: What does it mean to be for a value in Co or Contra- variant position?